### DISCUSSION TITLE
有关脚本总设置的问题

### DISCUSSION THREAD

**Discussion by timongh** [the1812/Bilibili-Evolved#4059] [提问 / Q&A]
1. 用户组件的选项同时维护了两份吗？为什么？
2. feature 的元数据在执行代码时就能拿到，为什么还要保存？
3. update 和 instance 是已经废弃的属性吗？

**Comment by the1812** [comment]
1. 一个是选项元数据, 一个是选项的用户保存值
2. 执行就是从保存的代码里读取啊
3. 是的

**Comment by timongh** [comment]
> 1. 用户组件的选项同时维护了两份吗？为什么？

我指的是这两个

<img src="https://user-images.githubusercontent.com/46739861/225501064-bd01a5b8-31dc-499d-8a8e-319556f3e676.png" width="400" />

**Comment by timongh** [comment]
> 2. feature 的元数据在执行代码时就能拿到，为什么还要保存？

我的意思是，既然在运行时从代码中拿 Metadata，那为什么还要存下来（如下图）？

<img src="https://user-images.githubusercontent.com/46739861/225502742-4fd48edf-b79f-40f7-88b5-1c8b39673146.png" width="400" />
